client/go/outline/method_channel.go

```diff
@@ -48,6 +75,14 @@ func InvokeMethod(method string, input string) *InvokeMethodResult {
 		}
 
 	default:
+		if h, ok := handlers[method]; ok {
```

Comment:
Why do you need this? Can't you just call the methods here directly?

The `outline` package should depend on other packages, but not the other way around.

Separating the methods will hurt readability, and having to register adds an extra step.

Reply:
Because Go complains about the dependency loop: `vpn` package needs to use `outline.Client`, and `outline.InvokeMethod` needs to use `vpn.EstablishVPN`. So a `vpn` -> `outline` -> `vpn` loop occurs.

Reply:
This is inverted though. The VPN api shouldn't depend on the application logic. The app depends on the VPN api.

The VPN API should declare its own inputs types (or reuse from a common library, like the SDK), and the app logic would use that. 

For example, you can make newVPNConnection take a routing config and a the transport object (StreamDialer+PacketListener?, Device? PacketProxy? You tell me). The app builds and passes the object. It's not clear to me what exactly which interface, but such interface must not be defined in the app logic.

Reply:
Removed.